Source/Details/ASPageTable.h

```diff
@@ -0,0 +1,82 @@
+//
+//  ASPageTable.h
+//  Texture
+//
+//  Copyright (c) 2017-present, Pinterest, Inc.  All rights reserved.
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+#import <AsyncDisplayKit/ASBaseDefines.h>
+
+@class ASCollectionElement;
+
+NS_ASSUME_NONNULL_BEGIN
+
+AS_SUBCLASSING_RESTRICTED
+@interface ASPageCoordinates : NSObject
+
+@property (nonatomic, assign, readonly) NSUInteger x;
+@property (nonatomic, assign, readonly) NSUInteger y;
+
++ (instancetype)pageCoordinatesWithX:(NSUInteger)x Y:(NSUInteger)y;
```

Comment:
True, an integer is preferable even to a struct, since it'll avoid malloc/free. That said, if the code is cleaner the code is cleaner and that's mostly what matters.